File: bandwidth.rb
#/usr/bin/ruby

require "csv"

file_name = "logs/openvpn-status.log"

table = CSV.read(file_name)

table.shift #discard first line

table.each do |row|
  if row[0] == "ROUTING TABLE"
    break
  end
  
  username = row[0]
  ip = row[1]
  bytes_received = row[2]
  bytes_sent = row[3]

# insert this stuff into a table or something
puts "User #{username} ip #{ip} received #{bytes_received} sent #{bytes_sent}"

end

File: bandwidth.py
#!/usr/bin/env python

import csv

vpn_status = csv.reader(open('openvpn-status.log', 'r'))

start_of_users = False
for row in vpn_status:
    if row[0] == 'Common Name':
        start_of_users = True
        continue
    if not start_of_users:
        continue
    if row[0] == 'ROUTING TABLE':
        break
    username = row[0]
    ip = row[1]
    bytes_received = row[2]
    bytes_sent = row[3]

    print "User %s ip %s received %s sent %s" % (username, ip, bytes_received, bytes_sent)
